src/main.rs

```diff
@@ -3039,23 +3039,19 @@ mod tests {
     }
 
     fn get_pss(pid: u32) -> u32 {
-        let output = Command::new("bash")
-            .arg("-c")
-            .arg(
-                format!(
-                    "cat /proc/{}/smaps | grep -i pss |  awk '{{Total+=$2}} END {{print Total}}'",
-                    pid
-                )
-                .as_str(),
-            )
-            .output()
-            .unwrap();
-
-        std::str::from_utf8(output.stdout.as_slice())
-            .unwrap()
-            .trim()
-            .parse::<u32>()
-            .unwrap()
+        let smaps = fs::File::open(format!("cat /proc/{pid}/smaps"), pid).unwrap();
+        let reader = std::io::BufReader::new(smaps);
+
+        let mut total = 0;
+        for line in reader.lines() {
+            // Lines look like this:
+            // Pss:                 176 kB
+            if line.contains("Pss") {
+                values = line.rsplit(" ").collect();
```

Comment:
Is the split going to work since there is more that one space betweem `Pss:` and the actual value?